superset/assets/visualizations/BigNumber.jsx

```diff
@@ -0,0 +1,184 @@
+  bottom: 4,
+  left: 4,
+};
+
+const NUMBER_SIZE_SCALAR = 0.3;
+const CHART_SIZE_SCALAR = 0.2;
+const TEXT_SIZE_SCALAR = 0.1;
+
+// Returns the maximum font size (<= idealFontSize) that will fit within the available Width
+function getMaxFontSize({ text, availableWidth, idealFontSize, fontWeight = 'normal' }) {
+  let fontSize = idealFontSize;
+  let textWidth = getTextWidth(text, `${fontWeight} ${fontSize}px ${fontFamily}`);
+  while (textWidth > availableWidth) {
+    fontSize -= 2;
+    textWidth = getTextWidth(text, `${fontWeight} ${fontSize}px ${fontFamily}`);
+  }
+  return fontSize;
+}
+
+function renderTooltipFactory({ formatDate, formatValue }) {
+  return function renderTooltip({ datum }) { // eslint-disable-line
+    const { x: rawDate, y: rawValue } = datum;
+    const formattedDate = formatDate(rawDate);
+    const value = formatValue(rawValue);
+
+    return (
+      <div style={{ padding: '4px 8px' }}>
+        {formattedDate}
+        <br />
+        <strong>{value}</strong>
+      </div>
+    );
+  };
+}
+function bigNumberVis(slice, payload) {
+  const { formData, containerId } = slice;
+  const json = payload.data;
+  const { data, subheader, compare_lag: compareLag, compare_suffix: compareSuffix } = json;
+  const showTrendline = formData.viz_type === 'big_number';
+  const formatValue = d3FormatPreset(formData.y_axis_format);
+  const formatPercentChange = d3.format('+.1%');
+  const formatDate = d3TimeFormatPreset('smart_date');
+  const renderTooltip = renderTooltipFactory({ formatDate, formatValue });
+  const gradientId = `big_number_${containerId}`;
+  const totalWidth = slice.width();
+  const totalHeight = slice.height();
+  const availableWidth = totalWidth - 2 * CONTAINER_PADDING;
+
+  const bigNumber = showTrendline ? data[data.length - 1][1] : data[0][0];
+  let percentChange = null;
+
+  if (showTrendline && compareLag > 0) {
+    const compareIndex = data.length - (compareLag + 1);
+    if (compareIndex >= 0) {
+      const compareValue = data[compareIndex][1];
+      percentChange = compareValue === 0
+        ? 0 : (bigNumber - compareValue) / Math.abs(compareValue);
+    }
+  }
+
+  const formattedBigNumber = formatValue(bigNumber);
+  const formattedData = showTrendline ? data.map(d => ({ x: d[0], y: d[1] })) : null;
+  const formattedSubheader = percentChange === null ? subheader : (
+    `${formatPercentChange(percentChange)} ${compareSuffix}`
+  );
+
+  const bigNumberFontSize = getMaxFontSize({
+    text: formattedBigNumber,
+    availableWidth,
+    idealFontSize: totalHeight * NUMBER_SIZE_SCALAR,
+    fontWeight: BIG_NUMBER_STYLES.fontWeight,
+  });
+
+  const subheaderFontSize = formattedSubheader ? getMaxFontSize({
+    text: formattedSubheader,
+    availableWidth,
+    idealFontSize: totalHeight * TEXT_SIZE_SCALAR,
+    fontWeight: SUBHEADER_STYLES.fontWeight,
+  }) : null;
+
+  const Visualization = (
+    <div style={{ height: totalHeight, ...CONTAINER_STYLES }}>
+      <div
+        style={{
```

Comment:
nit: I would use a const var or potentially `styles.json` file. I can see us have an `data-ui` directory just like the deckgl dir and creating components specifically around `data-ui` 

Would love to here your thoughts @williaster

Reply:
A `data-ui` dir is an interesting idea. I think a lot of these styles are still specific to big number, though, and are mostly targeted around dynamic font sizing. For instance here you can't use a const var beyond `BIG_NUMBER_STYLES` because the `fontSize` is dynamically computed